contrib/agent/src/main/resources/reference.conf

```diff
@@ -0,0 +1,21 @@
+# Reference configuration for the OpenCensus Agent for Java.
+
+opencensus.contrib.agent {
+
+  # Configuration settings related to automatic context propagation.
+  context-propagation {
+
+    # Enable/disable automatic context propagation for Executors.
+    executor = on
```

Comment:
is on syntax normal? just used to .enabled which maybe isn't idomatic in this library..

Reply:
Yes, `on` is one of the six strings (`true`/`false`, `on`/`off`, `yes`/`no`) that can be parsed to a boolean by typesafe config (https://github.com/lightbend/config/blob/master/HOCON.md#automatic-type-conversions).

Re `.enabled`: I haven't yet seen enough configs to make up my mind about this. It would definitely make sense if there are more details to configure, something like:
```
feature1.enabled = yes
feature1.host = www.example.com
feature1.port = 80
```
BTW, in HOCON, this is the same as:
```
feature1 {
  enabled = yes
  host = www.example.com
  port = 80
```
Now, considering that I later want to use this feature/detail pattern for configuring which (and how) methods will be instrumented, it makes sense to use the same pattern everywhere, if even there are no feature details to configure. Will update the PR later. (Ignoring nested features for now)